Priority Queue/FastPriorityQueue.cs

```diff
@@ -365,7 +407,8 @@ public void Remove(T node)
             #endif
 
             //If the node is already the last node, we can remove it immediately
-            if(node.QueueIndex == _numNodes)
+            int index = node.QueueIndex;
```

Comment:
Can you rename this to `removalIndex` or something more descriptive?

Also JFYI this makes no difference in the final assembly code, so I'd also be okay with removing it.